parser/statementStartTokens.coffee

```diff
@@ -0,0 +1,17 @@
+class StartTokens
+  @statement = [
+    'for', 'while',                                  # Start of statement
+    @expression
```

Comment:
This will be a nested array, which I don't think is what you intended it to be in your code.